File: go/server/service/book_service.go
package service

import (
	"github.com/csandiego/poc-openapitools/go/server/data"
)

type BookService interface {
	Create(data.Book) error
	Delete(int) error
	Get(int) (data.Book, error)
	List() ([]data.Book, error)
	Update(int, data.Book) error
}

File: swift/Client/Sources/Lib/BookService.swift
public protocol BookService {

    func list(completion: @escaping ((_ books: [Book]?, _ error: Error?) -> Void))

    func create(book: Book, completion: @escaping ((_ data: Void?, _ error: Error?) -> Void))

    func callGet(id: Int, completion: @escaping ((_ book: Book?, _ error: Error?) -> Void))

    func update(id: Int, book: Book, completion: @escaping ((_ data: Void?, _ error: Error?) -> Void))

    func delete(id: Int, completion: @escaping ((_ data: Void?, _ error: Error?) -> Void))
    
}
